osquery/logger/plugins/tests/buffered_tests.cpp

```diff
@@ -271,4 +276,74 @@ TEST_F(BufferedLogForwarderTests, test_split) {
       .WillOnce(Return(Status(0)));
   runner2.check();
 }
+
+// Test the purge() function independently of check()
+TEST_F(BufferedLogForwarderTests, test_purge) {
+  FLAGS_buffered_log_max = 3;
+  StrictMock<MockBufferedLogForwarder> runner("mock", kLogPeriod, 100);
+  for (uint64_t i = 0; i < 10; ++i) {
+    runner.logString(std::to_string(i));
+    StatusLogLine log1 = makeStatusLogLine(O_INFO, "foo", 1, "foo status");
+    StatusLogLine log2 = makeStatusLogLine(O_ERROR, "bar", 30, "bar error");
+    runner.logStatus({log1, log2});
+  }
+  // wait for a new timestamp
+  std::this_thread::sleep_for(std::chrono::milliseconds(1500));
+  runner.logString("foo");
+  runner.purge();
+  runner.logString("bar");
+  runner.purge();
+  runner.logString("baz");
+  runner.purge();
+  runner.purge();
+
+  EXPECT_CALL(runner, send(ElementsAre("foo", "bar", "baz"), "result"))
+      .WillOnce(Return(Status(0)));
+  runner.check();
+
+  runner.check();
```

Comment:
To verify that everything is properly cleared after the previous check. I'll add a comment to this effect.